### PR TITLE
[chore](cloud) Run GC if not enough memory when checkpoint

### PR DESCRIPTION
**Pull request by w41ter**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by w41ter** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -2841,6 +2841,14 @@ public static boolean isNotCloudMode() {
     @ConfField(description = {"存算分离模式下streamload导入使用的转发策略, 可选值为public-private或者空",
             "streamload route policy in cloud mode, availale options are public-private and empty string"})
     public static String streamload_redirect_policy = "";
+
+    // The count threshold to do manual GC when doing checkpoint but not enough memory.
+    // The default checkpoint interval is 1min.
+    @ConfField(mutable = true, description = {
+            "存算分离模式下如果 checkpoint 连续多次因内存不足而无法进行时，先尝试手动触发 GC",
+            "The threshold to do manual GC when doing checkpoint but not enough memory"})
+    public static int cloud_checkpoint_manual_gc_threshold = 10;
```
Comment (gavinchou) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2850]: The `cloud_` prefix can be omitted

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by w41ter** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**